xcat-inventory/xcclient/inventory/manager.py

```diff
@@ -99,100 +127,160 @@ def importObjs(self, objlist, obj_attr_dict,update=True):
+            raise CommandException("Error: The specified inventory directory does not exist: %(p)s", p=args.path)
 
     if action == 'export': #extra validation for export
+        #if args.path and (not args.type or args.type !='osimage'):
+        #    raise CommandException("Error: -f|--path is only valid for -t|--type=osimage")
         if args.format and args.format.lower() not in VALID_OBJ_FORMAT:
             raise CommandException("Error: Invalid exporting format: %(f)s", f=args.format)
         if args.exclude:
             for et in args.exclude.split(','):
                 if et.lower() not in InventoryFactory.getvalidobjtypes():
                     raise CommandException("Error: Invalid object type to exclude: \"%(t)s\"", t=et)
 
-def export_by_type(objtype, names, location, fmt,version=None):
+def export_by_type(objtype, names, destfile=None, destdir=None, fmt='json',version=None,exclude=None):
+    if objtype and objtype != 'osimage' and destdir:
+        raise CommandException("Error: directory %(f)s specified by -f|--path is only supported when [-t|--type osimage] or export all without [-t|--type] specified",f=destdir)
+
     InventoryFactory.getLatestSchemaVersion()
     dbsession=DBsession()
-    hdl = InventoryFactory.createHandler(objtype,dbsession,version)
-    objlist = []
-    if names:
-        objlist.extend(names.split(','))
 
-    typedict=hdl.exportObjs(objlist, location)
-    nonexistobjlist=list(set(objlist).difference(set(typedict[objtype].keys())))
-    if nonexistobjlist:
-        raise ObjNonExistException("Error: cannot find objects: %(f)s!", f=','.join(nonexistobjlist))
-   
-    if version is None or version in ('latest'):
-        version=InventoryFactory.getLatestSchemaVersion()
-    typedict['schema_version']=version    
-    
-    if not fmt or fmt.lower() == 'json':
-        dump2json(typedict)
+    objlist = []
+    objtypelist=[]
+    exportall=0
+    if objtype:
+        objtypelist.extend(objtype.split(','))
     else:
-        dump2yaml(typedict)
-    dbsession.close() 
+        objtypelist.extend(InventoryFactory.getvalidobjtypes())
+        exportall=1
 
+    if names:
+        objlist.extend(names.split(','))
 
-def export_all(location, fmt, exclude=None, version=None):
-    dbsession=DBsession()
-    wholedict={}
-    for objtype in InventoryFactory.getvalidobjtypes():
-        if exclude and objtype in exclude:
+    wholedict={} 
+    for myobjtype in objtypelist:
+        if exclude and myobjtype in exclude:
             continue
-        hdl = InventoryFactory.createHandler(objtype,dbsession,version)
-        wholedict.update(hdl.exportObjs([]))
+        hdl = InventoryFactory.createHandler(myobjtype,dbsession,version)
+        if myobjtype == 'osimage' and destdir:
+            if exportall:
+                mylocation=destdir+'/'+myobjtype
+                if os.path.exists(mylocation):
+                    shutil.rmtree(mylocation)
```

Comment:
why we need to delete the directory, it will wipe out the .git if user want to export it to a git directory.